compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1580,4 +1580,10 @@ object messages {
     val explanation = ""
   }
 
+  case class SuperCallsNotAllowedInline(symbol: Symbol)(implicit ctx: Context)
+    extends Message(SuperCallsNotAllowedInlineID) {
+    val kind = "Syntax"
+    val msg = s"super call not allowed in inline $symbol"
+    val explanation = ""
```

Comment:
Maybe it would be good to have an explanation here on why `inline` inherently disallows super calls.